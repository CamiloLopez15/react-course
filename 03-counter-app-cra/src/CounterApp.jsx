import PropTypes from "prop-types";
import React from "react";

function CounterApp({ counter }) {
    return (
        <>
            <h1>CounterApp</h1>
            <span>/{counter}</span>
        </>
    );
}

export default CounterApp;

CounterApp.propTypes = {
    counter: PropTypes.number.isRequired,
};
