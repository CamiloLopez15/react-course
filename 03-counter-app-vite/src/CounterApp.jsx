import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 1);
    };

    const handleSubstract = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <h2> {counter} </h2>

            <button onClick={handleAdd} data-testid="test-button-+1">
                {" "}
                +1{" "}
            </button>
            <button onClick={handleSubstract} data-testid="test-button--1">
                {" "}
                -1{" "}
            </button>
            <button
                onClick={handleReset}
                aria-label="test-button-reset"
            >
                {" "}
                Reset{" "}
            </button>
        </>
    );
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};
