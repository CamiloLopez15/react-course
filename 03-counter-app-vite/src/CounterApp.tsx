import { useState } from "react";

function CounterApp({ value }: { value: number }) {
    const [counter, setCounter] = useState<number>(value);

    const handleAdd = () => setCounter((beforeValue) => beforeValue + 1);
    const handleSubtract = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);

    return (
        <>
            <h1>CounterApp</h1>
            <span>{counter}</span><br />
            <button onClick={handleAdd}>+ 1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSubtract}>- 1</button>
        </>
    );
}

export default CounterApp;
