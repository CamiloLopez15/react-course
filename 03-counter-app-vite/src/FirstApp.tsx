export default function FirstApp({
    title,
    subTitle,
}: {
    title: string;
    subTitle: string;
}) {
    console.log({ title, subTitle });

    return (
        <>
            <h1>First App</h1>
            <p>Hola mundo</p>
        </>
    );
}
