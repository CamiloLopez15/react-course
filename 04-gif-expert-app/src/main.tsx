import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import GifExpertApp from "./page/GifExpertApp";

createRoot(document.getElementById("root")!).render(
    <StrictMode>
        <GifExpertApp />
    </StrictMode>
);
