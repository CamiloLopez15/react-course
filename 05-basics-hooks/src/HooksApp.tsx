function HooksApp() {
    return <h1>Hooks Appcvzxcvzxcvzxcvzxcv</h1>;
}

export default HooksApp;
