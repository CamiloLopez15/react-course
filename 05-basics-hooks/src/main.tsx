import "./index.css";
import { StrictMode } from "react";
import { createRoot } from "react-dom/client";
import TodoApp from "./08-useReducer/TodoApp";
// import CounterApp from "./01-useState/CounterApp";
// import HooksApp from "./HooksApp";
// import CounterWithCustomHook from "./01-useState/CounterWithCustomHook";
// import SimpleForm from "./02-useEffect/SimpleForm";
// import FormWithCustomHook from "./02-useEffect/FormWithCustomHook";
// import { MultipleCustomHooks } from "./03-example/MultipleCustomHooks";
// import { FocusScreen } from "./04-useRef/FocusScreen";
// import { Memorize } from "./06-memos/Memorize";
// import { MemoHook } from "./06-memos/MemoHook";
// import { CallBackHook } from "./06-memos/CallbackHook";
// import { Padre } from "./07-homework/Padre";
// import "./08-useReducer/intro-reducer";

createRoot(document.getElementById("root")!).render(
    <StrictMode>
        <TodoApp />
    </StrictMode>
);
