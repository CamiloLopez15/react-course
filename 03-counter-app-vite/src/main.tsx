import { createRoot } from "react-dom/client";
import { StrictMode } from "react";
// import FirstApp from "./FirstApp";
import "./styles.css";
import CounterApp from "./CounterApp";

createRoot(document.getElementById("root")!).render(
    <StrictMode>
        {/* <App /> */}
        {/* <FirstApp title="Titulo" subTitle="Subtitulo"/>
         */}
        <CounterApp value={10} />
    </StrictMode>
);
