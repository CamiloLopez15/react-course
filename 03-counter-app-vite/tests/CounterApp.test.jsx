import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Pruebas en el componente CounterApp", () => {
    const initialValue = 100;
    test("El componente debe hacer  match con el snapshot", () => {
        const { container } = render(<CounterApp value={initialValue} />);
        expect(container).toMatchSnapshot();
    });
    test(`Debe mostrar el valor inicial en ${initialValue}`, () => {
        render(<CounterApp value={initialValue} />);
        expect(screen.getByText(initialValue)).toBeTruthy();
        //? OR
        expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
            String(initialValue)
        );
    });
    test("Debe incrementar con el botón +1", () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByTestId("test-button-+1"));
        expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
            String(initialValue + 1)
        );
    });
    test("Debe de decrementar con el botón -1", () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByTestId("test-button--1"));
        expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
            String(initialValue - 1)
        );
    });
    test("Debe de resetear valor con el botón reset", () => {
        render(<CounterApp value={initialValue} />);
        fireEvent.click(screen.getByTestId("test-button-+1"));
        fireEvent.click(screen.getByTestId("test-button-+1"));
        fireEvent.click(screen.getByTestId("test-button-+1"));
        fireEvent.click(screen.getByTestId("test-button-+1"));
        fireEvent.click(screen.getByTestId("test-button-+1"));
        fireEvent.click(
            screen.getByRole("button", { name: "test-button-reset" })
        );
        expect(screen.getByRole("heading", { level: 2 }).innerHTML).toContain(
            String(initialValue)
        );
    });
});
