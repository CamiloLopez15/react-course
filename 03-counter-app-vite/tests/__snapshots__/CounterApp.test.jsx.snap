// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CounterApp El componente debe hacer  match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    data-testid="test-button-+1"
  >
     
    +1
     
  </button>
  <button
    data-testid="test-button--1"
  >
     
    -1
     
  </button>
  <button
    aria-label="test-button-reset"
  >
     
    Reset
     
  </button>
</div>
`;
